### PR TITLE
[serve] Remove previously deprecated API options

### PR DESCRIPTION
**Pull request by edoakes**
## Why are these changes needed?

- `host` and `port` in `serve.run` and `serve run`
- `dedicated_cpu` and `detached` in `serve.start`

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: python/ray/serve/api.py [hunk 1]
```diff
@@ -299,8 +278,6 @@ class MyDeployment:
             this deployment. Defaults to 1.
         autoscaling_config: Parameters to configure autoscaling behavior. If this
             is set, `num_replicas` cannot be set.
-        route_prefix: [DEPRECATED] Route prefix should be set per-application
```
Comment (zcin) [inline python/ray/serve/api.py:302]: If we are removing `route_prefix` from @serve.deployment, should we also remove from `.options()`? Also, I think there are still a few stray tests that use `route_prefix`.

Comment (edoakes) [reply python/ray/serve/api.py:302]: Yes we should... that might be more involved. Let me revert the route_prefix part here and can you make a separate PR to remove those?

Comment (zcin) [reply python/ray/serve/api.py:302]: Yeah, I can handle removing route  prefix.

**Comment by edoakes** [comment]
@zcin can I get a re-review/approval here please?

**Review by zcin — APPROVED** [review]
LGTM!

**Comment by name-snrl** [comment]
@khluu I'm sorry for bothering you, but how I should configure host/port now when use `serve.run()`?

**Comment by name-snrl** [comment]
@edoakes

**Pull request merged**